<a href="https://colab.research.google.com/github/TianaT6/MSSP607/blob/main/Modules/Week_4/MSSP607_Week4_Participaton.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# MSSP607 Week 4

I used all rows in `parsed.csv` for this activity.


### 1.a

In [12]:
import pandas as pd

import pandas as pd

df = pd.read_csv("/parsed.csv")
df[["mag", "magType", "parsed_place", "tsunami"]].head()

,mag,magType,parsed_place,tsunami
0,1.35,ml,California,0
1,1.29,ml,California,0
2,3.42,ml,California,0
3,0.44,ml,California,0
4,2.16,md,California,0


### 2.a


In [13]:
japan_mb = df[(df["parsed_place"] == "Japan") & (df["magType"] == "mb")]
percentile_95 = japan_mb["mag"].quantile(0.95)
print("95th percentile:", percentile_95)


95th percentile: 4.9


The 95th percentile of magnitude in Japan for `mb` is **4.9**.


### 2.b


In [14]:
indonesia = df[df["parsed_place"] == "Indonesia"]
tsunami_count = indonesia["tsunami"].sum()
tsunami_percentage = tsunami_count / len(indonesia) * 100
print("Tsunami flags:", tsunami_count)
print("Total Indonesia records:", len(indonesia))
print(f"Percentage: {tsunami_percentage:.2f}%")


Tsunami flags: 34
Total Indonesia records: 147
Percentage: 23.13%


**34 out of 147** records in Indonesia have a tsunami flag. The percentage is **23.13%**.


### 2.c


In [15]:
nevada = df[df["parsed_place"] == "Nevada"]
nevada.describe()


,cdi,dmin,felt,gap,mag,mmi,nst,rms,sig,time,tsunami,tz,updated
count,15.000000,681.000000,15.000000,681.000000,681.000000,1.00,681.000000,681.000000,681.000000,6.810000e+02,681.0,681.0,6.810000e+02
mean,2.440000,0.166199,2.400000,153.668120,0.500073,2.84,12.618209,0.151986,10.970631,1.538314e+12,0.0,-480.0,1.538402e+12
std,0.501142,0.166228,4.626013,68.735302,0.696710,NaN,9.866963,0.084662,19.607150,5.965637e+08,0.0,0.0,6.010951e+08
min,2.000000,0.001000,1.000000,29.140000,-0.500000,2.84,3.000000,0.000500,0.000000,1.537247e+12,0.0,-480.0,1.537307e+12
25%,2.000000,0.053000,1.000000,97.380000,-0.100000,2.84,6.000000,0.106900,0.000000,1.537854e+12,0.0,-480.0,1.537928e+12
50%,2.200000,0.112000,1.000000,149.140000,0.400000,2.84,10.000000,0.146300,2.000000,1.538280e+12,0.0,-480.0,1.538428e+12
75%,2.900000,0.233000,1.000000,199.720000,0.900000,2.84,16.000000,0.187100,12.000000,1.538821e+12,0.0,-480.0,1.538878e+12
max,3.300000,1.414000,19.000000,355.910000,2.900000,2.84,61.000000,0.863400,129.000000,1.539461e+12,0.0,-480.0,1.539483e+12


Nevada has **681** records. The mean magnitude is about **0.50**, and the median is **0.40**. The minimum is **-0.50**, and the maximum is **2.90**.


### 2.d


In [16]:
ring_locations = [
    "Bolivia", "Chile", "Ecuador", "Peru", "Costa Rica", "Guatemala",
    "Mexico", "Japan", "Philippines", "Indonesia", "New Zealand",
    "Canada", "Fiji", "Alaska", "Washington", "California", "Russia",
    "Taiwan", "Tonga", "Kermadec Islands"
]

df["ring_of_fire"] = (
    df["parsed_place"].isin(ring_locations)
    | df["parsed_place"].str.contains("Antarctic", na=False)
)
df[["parsed_place", "ring_of_fire"]].head()


,parsed_place,ring_of_fire
0,California,True
1,California,True
2,California,True
3,California,True
4,California,True


I added `ring_of_fire` to mark the listed locations. Exact matching excludes New Mexico, and the Antarctic search includes the Antarctic locations.


### 2.e


In [17]:
inside_count = df["ring_of_fire"].sum()
outside_count = (~df["ring_of_fire"]).sum()
print("Inside Ring of Fire:", inside_count)
print("Outside Ring of Fire:", outside_count)


Inside Ring of Fire: 7188
Outside Ring of Fire: 2144


There are **7,188** records inside the listed Ring of Fire locations and **2,144** outside them.


### 2.f


In [18]:
ring_tsunami_count = df.loc[df["ring_of_fire"], "tsunami"].sum()
print("Tsunami flags along the Ring of Fire:", ring_tsunami_count)


Tsunami flags along the Ring of Fire: 45


There are **45** tsunami flags in the Ring of Fire locations.
